Make the free plugin work better with the Pro one

kienstra opened the pull request:
Updated 23 August 

#### Changes
* Adds a [script](https://github.com/studiopress/genesis-custom-blocks/pull/18/files#r475154858) to create a built tag that the Pro plugin can require in `composer.json`
* Changes some hooks 
* Adds CSS linting

#### Testing instructions
1. Create a block with all 13 free fields 
2. Expected: They still work as expected, and the `<ServerSideRender>` displays them in the block editor
3. Expected: None of the 6 Pro fields is available: Post, Taxonomy, User, Rich Text, Classic Text, Repeater

kienstra committed:
Don't show the Genesis Pro upgrade page on GCB Pro

The user would already have
GCB Pro, so no need for an upsell.

kienstra committed:
Fix an issue in displaying the Pro nag

It should only show
if the filter returns true.

kienstra committed:
Add a deleted empty line

kienstra committed:
Add spaces between style rules

kienstra committed:
Remove styling from the Pro plugin

I should have removed this before,
as it's for Pro controls.

kienstra committed:
Add a CSS linting script from Gutenberg

And do npm run lint:css:fix
to apply it.

kienstra committed:
Simplify the documentation of a filter

2 DocBlocks aren't needed
for 2 calls of apply_filters().

kienstra committed:
Add a method get_controls() that the Pro plugin or any other can use

Needs to be called after 'init',
so the controls were registered.

kienstra committed:
Add a filter that the Pro plugin can use

This is really messy,
there might be a better way.
But ideally the React refactor will
remove this.

kienstra committed:
Remove the reference to the Code of Conduct

In CONTRIBUTING.md

Because it references info@getblocklab.com,

kienstra committed:
Remove a seemingly duplicate hook

It looks like this output
the same markup that was already there.

kienstra committed:
Change how this gets the field type

Instead of instantiating a new control,
use one that is already in BlockPost.

kienstra commented on CONTRIBUTING.md:
```diff
@@ -21,10 +19,6 @@ The following is a set of guidelines for contributing to Genesis Custom Blocks.
-## Code of Conduct
-
-This project and everyone participating in it is governed by the [Genesis Custom Blocks Code of Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code.  Please report unacceptable behavior to [info@getblocklab.com](mailto:info@getblocklab.com).
-
```
We could keep this part about the Code of Conduct, but what email should we use?

mindctrl replied:
@mattlawck is there an official Code of Conduct policy? I saw a few in other WPE repos, but they looked more generic and didn't include any contact info.

kienstra replied:
Ah, thanks

kienstra committed:
Change the slug of the 'block-post' script

Simply 'block-post' is probably too generic,

so prefix it with the plugin name.

kienstra committed:
Update CONTRIBUTING.md for phpunit

Instructions about setting up PHPUnit
and running it

kienstra committed:
Attempt to address failed Composer install

Remove the previous phpunit
before installing the new one.

kienstra committed:
Remove phpunit from dev

kienstra committed:
In Composer, remove phpunit/phpunit before doing composer install

kienstra committed:
Add an automated way to tag a built plugin, mainly from AMP plugin

Thanks to Weston Ruter
for his work on this.

kienstra committed:
In composer install, don't build dev dependencies

kienstra committed:
Move gulp higher so it has access to version

kienstra committed:
Minor improvements to tag-built.sh syntax

kienstra committed:
Remove the exit if in dirty state

It usually exits,
but I don't see a problem in
the git state.

kienstra committed:
Add back in a missing template

This didn't allow adding
in a new field.

kienstra committed:
Rename the template, as it's not for a repeater field

Rename to gcb-field,
as it's for a field.

kienstra committed:
Remove phpunit/phpunit in e2e test

It looks to be failing because it
can't resolve all of the dependencies.

kienstra committed:
Move the icon's inline styles to attributes

Instead of inline styles,
have individual attributes.

kienstra committed:
Add a Release Procedure to CONTRIBUTING.md

kienstra committed:
Remove an extra Node line, and combine 2 lines into 1

kienstra commented on .eslintignore:
```diff
@@ -1,2 +1 @@
 js/editor.blocks.js
-js/scripts.js
```
This file doesn't exist 😄

kienstra commented on css/admin.settings.css:
```diff
@@ -1,3 +0,0 @@
-.genesis-custom-blocks-settings .nav-tab-wrapper .dashicons-before:before {
-	padding: 2px 3px 0 0;
-}
```
The Settings page isn't in this free plugin anymore

kienstra commented on css/src/editor.scss:
```diff
@@ -1,4 +1,4 @@
-@import 'color';
+@import "color";
```
A lot of this diff is from the new command `npm run lint:css:fix`.

There are also style rules moved to the Pro plugin, like for Pro controls.

kienstra commented on js/admin.block-post.js:
```diff
@@ -15,7 +15,7 @@
 
 		$( '#block-add-field' ).on( 'click', function() {
-			const template = wp.template( 'field-repeater' ),
+			const template = wp.template( 'gcb-field' ),
```
The name `field-repeater` was a little confusing, as it wasn't for the Repeater field. It made sense before we added the Repeater 😄

kienstra commented on tests/wp-tests-config.php:
```diff
@@ -1,39 +0,0 @@
-<?php
```
This shouldn't be needed anymore, it's from when we used Travis.

kienstra committed:
Remove the functions to check versions

Core does this on plugin installation,
with validate_plugin_requirements().

kienstra committed:
Install dependencies as part of gulp

So this should only require running gulp,
no steps before it.

kienstra committed:
Move gulp command lower, as it was a little hidden

kienstra committed:
Fork a script to verify versions from Weston's work

From the Official AMP Plugin for WordPress.
And run this in a gulp task.

kienstra committed:
Add a final newline at the end of output.

kienstra committed:
Merge in develop, attempt to resolve conflict

This branch deleted wp-tests-config.php,
while the develop branch has an edit of it.

kienstra committed:
Again attempt to resolve conflict

kienstra commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -18,7 +18,10 @@ about: Report a problem
+<!-- Please include screenshots of the behavior --> 
+
+## JS Console
+<!-- Please copy the JS console here -->
```
We usually need the JS console in bug reports.

kienstra commented on CONTRIBUTING.md:
```diff
@@ -2,104 +2,52 @@
+
+### Release Procedure
+
+1. `checkout` locally whatever branch you want to release. It could be `develop`, or a release branch like `1.0`. 
```
The 'Release Procedure' is definitely a work in progress, feel free to suggest anything.

kienstra commented on assets/images/admin-menu-icon.svg:
```diff
@@ -1,5 +1,3 @@
-		<path style=" stroke:none;fill-rule:evenodd;fill:rgb(0%,0%,0%);fill-opacity:1;" d="M 11.421875 10.09375 C 11.5 10.296875 11.605469 10.472656 11.738281 10.613281 C 11.871094 10.753906 12.03125 10.863281 12.214844 10.9375 C 12.398438 11.011719 12.601562 11.046875 12.820312 11.046875 C 12.984375 11.046875 13.132812 11.035156 13.261719 11.007812 C 13.386719 10.980469 13.507812 10.941406 13.625 10.8 ...
-	</g>
+<svg fill-rule="evenodd" clip-rule="evenodd" stroke-linecap="round" stroke-linejoin="round" stroke-miterlimit="2" width="100%" height="100%" viewBox="0 0 82 68" xmlns="http://www.w3.org/2000/svg">
+	<path fill="#82878c" id="genesis_icon_small" d="M77.71,37.138c3.429,3.7 4.972,7.309 4.007,10.359c-1.128,3.565 -5.495,5.788 -11.872,6.618c-6.155,8.104 -15.891,13.34 -26.855,13.34c-15.613,0 -28.735,-10.618 -32.565,-25.026c9.417,6.084 19.646,8.178 31.871,9.842c-1.955,-0.506 -3.936,-1.071 -5.936,-1.704c-10.934,-3.459 -20.353,-8.193 -26.868,-13.107c-0.433,-0.328 -0.853,-0.654 -1.261,-0.981l-0.002,-0. ...
```
The 'G' is bigger now:

<img width="222" alt="g-is-bigger" src="https://user-images.githubusercontent.com/4063887/90968334-bdeca100-e4b0-11ea-8f8c-e0dd446dc644.png">

kienstra commented on bin/tag-built.sh:
```diff
@@ -0,0 +1,36 @@
+
+set -e
+
+gulp
```
The Pro plugin needs a built version of this free plugin to require in `composer.json`.

So this builds the plugin and creates a tag from the build.

kienstra replied:
Props Weston 😄

kienstra commented on genesis-custom-blocks.php:
```diff
@@ -27,67 +27,6 @@
-/**
- * Admin notice for incompatible versions of PHP.
- */
-function genesis_custom_blocks_php_version_error() {
```
Core checks the PHP and WP versions in [validate_plugin_requirements()](https://github.com/WordPress/wordpress-develop/blob/f556437ef433cacb67299ec3ee40ce56311077da/src/wp-admin/includes/plugin.php#L1129). So maybe we can keep this file a bit cleaner 😄

kienstra commented on php/Admin/Admin.php:
```diff
@@ -54,17 +54,8 @@ public function init() {
-			$this->upgrade = new Upgrade();
-			genesis_custom_blocks()->register_component( $this->upgrade );
-		}
+		$this->upgrade = new Upgrade();
```
Pro users will never see the upgrade page, as it has:

```php
add_filter( 'genesis_custom_blocks_show_pro_nag', '__return_false' );
```

kienstra committed:
Add a test script to composer.json

Now, you can run phpunit with
composer test.

kienstra committed:
Update the WordPress npm packages

Do npm run packages-update,
and fix the JS linting.

kienstra committed:
Update all but 1 npm package in package.json

From doing npm outdated,
update all but puppeteer.

kienstra committed:
Fix the dependency-extraction-webpack-plugin path

kienstra committed:
Revert "Again attempt to resolve conflict"

This reverts commit 15704549ad30a633125af74397f6bd204306a2b3.

kienstra committed:
Get the wp-tests-config.php file from develop

kienstra committed:
Change action name to genesis_custom_blocks_render_field_templates

This is actually a place to render templates,
not actions.

kienstra committed:
Use the Gutenberg utilty method insertBlock()

kienstra committed:
Add a line about doing composer install after gulp

kienstra committed:
Add placeholder URLs to get Genesis Pro

Maybe these should be changed,
but for now there are URLs.

kienstra commented on php/Views/Upgrade.php:
```diff
@@ -7,13 +7,15 @@
  * @license   http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
  */
 
+$genesis_pro_url = 'https://my.wpengine.com/signup?plan=genesis-pro';
```
Is this URL alright for the 'Get Genesis Pro' and 'Learn More' links below?

<img width="851" alt="get-g-pro" src="https://user-images.githubusercontent.com/4063887/90972629-577f7700-e4e0-11ea-9044-e59b522595e1.png">

<img width="1048" alt="the-links-learn-more" src="https://user-images.githubusercontent.com/4063887/90972618-433b7a00-e4e0-11ea-8213-aa4d889e190e.png">

lukecarbis replied:
The bottom two button links should point to: https://www.studiopress.com/genesis-pro/

kienstra replied:
OK, thanks!

kienstra committed:
Pass controls to filter, so extending plugin can use it

kienstra committed:
Fix failed Jest unit test

kienstra committed:
Remove quotes around ls-files

This has caused an error,
so remove them.

kienstra committed:
Move the gulp command lower

kienstra committed:
Revert "Move the gulp command lower"

This reverts commit 491a80e59fa2f61c75b9cc39f3b246109a4e55fa.

kienstra commented:
Hi @lukecarbis,
Hope you had a great weekend.

Could you please review this? The [testing instructions](https://github.com/studiopress/genesis-custom-blocks/pull/18#issue-467022181) are up to date.

Thanks, Luke!

lukecarbis commented on css/admin.upgrade.css:
```diff
@@ -63,6 +70,7 @@
 	border-radius: 3px;
 }
+
 .genesis-custom-blocks-pro .btn {
```
```suggestion
.genesis-custom-blocks-pro .container .btn {
```

This will help fix the styling of the two lower buttons.

lukecarbis commented on css/admin.upgrade.css:
```diff
@@ -84,19 +92,23 @@
 	color: var(--color-brand);
 }
+
 .genesis-custom-blocks-pro .btn-secondary {
```
```suggestion
.genesis-custom-blocks-pro .container .btn-secondary {
```

This will help fix the styling of the two lower buttons.

lukecarbis commented on css/admin.upgrade.css:
```diff
@@ -84,19 +92,23 @@
 	transition-duration: .2sec;
 }
+
 .genesis-custom-blocks-pro .btn:hover {
```
```suggestion
.genesis-custom-blocks-pro .container .btn:hover {
```

This is just for consistency.

kienstra replied:
Sounds good

lukecarbis commented on css/admin.upgrade.css:
```diff
@@ -84,19 +92,23 @@
 	border-color: var(--color-brand);
 }
+
 .genesis-custom-blocks-pro .btn-secondary:hover {
```
```suggestion
.genesis-custom-blocks-pro .container .btn-secondary:hover {
```

This is just for consistency.

lukecarbis commented on CONTRIBUTING.md:
```diff
@@ -2,104 +2,53 @@
+### Release Procedure
+
+1. `checkout` locally whatever branch you want to release. It could be `develop`, or a release branch like `1.0`. 
+1. Do `gulp`, and you'll see a `genesis-custom-blocks.zip` file in this plugin directory.
```
See one of the suggestions here: https://github.com/studiopress/genesis-custom-blocks-pro/pull/1#pullrequestreview-473055317

Could we move this to `package` directory?

kienstra replied:
Sure, good idea

kienstra replied:
It's moved in https://github.com/studiopress/genesis-custom-blocks/pull/18/commits/ac6f3218ae5550d9cd27500d83417a6b79cc23d8

lukecarbis reviewed (changes requested):
This is looking great @kienstra. Couple of really small suggestions.

mindctrl commented on .circleci/config.yml:
```diff
@@ -75,7 +75,7 @@ jobs:
           working_directory: *PLUGIN_PATH
           command: |
-            composer install && npm install && npm run build
+            composer remove --dev phpunit/phpunit && composer install && npm install && npm run build
```
@kienstra on these `npm install` steps... thoughts on switching them to `npm ci` instead? That would ensure the packages installed and used during the build process match the ones in the lock file, as opposed to some of them possibly being updated and different from what you and others have used and tested locally.

kienstra replied:
Ah. good idea. I'll think about that and respond here.

kienstra replied:
That's a good idea, I'll change it to `npm ci`.

kienstra replied:
It's changed in https://github.com/studiopress/genesis-custom-blocks/pull/18/commits/ab549624a5bb64c804a8256fb7808f8490dbe605

kienstra committed:
Commit Luke's GitHub suggestion for css/admin.upgrade.css

Co-authored-by: Luke Carbis <luke@carb.is>

kienstra committed:
Commit Luke's GitHub suggestion for css/admin.upgrade.css

Co-authored-by: Luke Carbis <luke@carb.is>

kienstra committed:
Commit Luke's GitHub suggestion for css/admin.upgrade.css

Co-authored-by: Luke Carbis <luke@carb.is>

kienstra committed:
Commit Luke's GitHub suggestion for css/admin.upgrade.css

Co-authored-by: Luke Carbis <luke@carb.is>

kienstra committed:
Change npm install to npm ci, thanks to John

Thanks to John for the suggestion.
As he mentioned, this ensures that
the packages match the lock file versions.

kienstra committed:
Create the built  file in package/, not the plugin root

And update CONTRIBUTING.md
for this change.

kienstra committed:
Add built/ to .gitinore, ensure it's not in the plugin .zip

kienstra committed:
Change the 'Learn more' links to the URL that Luke mentioned

lukecarbis reviewed (approved):
Looking good @kienstra! Plugin works, and builds, nicely.

kienstra commented:
Thanks, Luke!
